src/rpp/rpp/schedulers/trampoline_scheduler.hpp

```diff
@@ -0,0 +1,92 @@
+//                  ReactivePlusPlus library
+//
+//          Copyright Aleksey Loginov 2022 - present.
+//                            TC Wang 2022 - present.
+// Distributed under the Boost Software License, Version 1.0.
+//    (See accompanying file LICENSE_1_0.txt or copy at
+//          https://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/victimsnino/ReactivePlusPlus
+//
+
+#pragma once
+
+#include <rpp/schedulers/fwd.hpp>                       // own forwarding
+#include <rpp/schedulers/details/worker.hpp>            // worker
+#include <rpp/subscriptions/composite_subscription.hpp> // lifetime
+#include <rpp/schedulers/details/queue_worker_state.hpp>// state
+
+#include <concepts>
+#include <chrono>
+#include <functional>
+#include <thread>
+
+thread_local bool queue_in_use = false;
+thread_local rpp::schedulers::details::queue_worker_state queue{};
```

Comment:
@victimsnino I found this is the easiest way to create thread-local queues but I'm not sure if it's the right way. Besides, I'm seeing compile error about duplicate symbols related to this.

Reply:
> I'm seeing compile error about duplicate symbols related to this

Yes, because you've created global symbols and each compilation unit which includes this would try to create this one. Place them inside scheduler body

Reply:
Discussed offline: we use a static class function to house static `thread_local` variables.